test/e2e/cluster_authorized_cidrs_connectivity.go

```diff
@@ -0,0 +1,466 @@
+
+				By("testing connectivity from current machine (should be blocked)")
+				// Try to connect from the test runner (which is not in authorized CIDRs)
+				err = testAPIConnectivity(apiURL, 5*time.Second)
+				if err != nil {
+					By(fmt.Sprintf("Connection from unauthorized IP correctly blocked: %v", err))
+				} else {
+					// If we can connect, it means the test runner's IP might be in the cluster's network
+					// This is expected in some scenarios, so we just log it
+					By("Warning: Connection from test runner succeeded - this may indicate the runner is in the authorized network")
+				}
+
+				By("verifying VM can access cluster API with credentials")
+				adminRESTConfig, err := framework.GetAdminRESTConfigForHCPCluster(
+					ctx,
+					tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient(),
+					*resourceGroup.Name,
+					clusterName,
+					10*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Create kubeconfig and copy to VM
+				kubeconfig, err := generateKubeconfig(adminRESTConfig)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Test kubectl command from VM
+				// Use base64 encoding to safely transfer kubeconfig
+				kubeconfigB64 := base64.StdEncoding.EncodeToString([]byte(kubeconfig))
+				kubectlTest := fmt.Sprintf("echo '%s' | base64 -d > /tmp/kubeconfig && kubectl --kubeconfig=/tmp/kubeconfig get nodes 2>&1", kubeconfigB64)
+				output, err = runVMCommand(ctx, tc, *resourceGroup.Name, vmName, kubectlTest)
+				By(fmt.Sprintf("kubectl output from authorized VM: %s", output))
+
+				// Should be able to run kubectl commands (even if nodes aren't ready)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(output).To(Or(
+					ContainSubstring("NAME"),         // Success - got nodes
+					ContainSubstring("No resources"), // Success - no nodes yet
+					ContainSubstring("NotReady"),     // Success - nodes not ready
+				), "Should be able to execute kubectl from authorized VM")
+
+				By("updating cluster to remove VM from authorized CIDRs")
+				// Get the current cluster state
+				currentCluster, err := framework.GetHCPCluster(
+					ctx,
+					tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient(),
+					*resourceGroup.Name,
+					clusterName,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Update the cluster's authorized CIDRs
+				currentCluster.Properties.API.AuthorizedCIDRs = []*string{
+					to.Ptr("192.0.2.0/24"), // Use TEST-NET-1 (reserved for documentation)
+				}
+
+				// Use CreateOrUpdate (PUT) to apply the change
+				poller, err := tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient().BeginCreateOrUpdate(
+					ctx,
+					*resourceGroup.Name,
+					clusterName,
+					currentCluster.HcpOpenShiftCluster,
+					nil,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				_, err = poller.PollUntilDone(ctx, nil)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Wait a bit for the change to propagate
+				time.Sleep(30 * time.Second)
```

Comment:
Let's poll until a timeout instead of `sleep`, which introduces race conditions.

Reply:
I think the intent is to put the below check in a retry loop that sleeps, say, 5 seconds between tries and retries up to 6 times.  There's your 30 seconds.  Or adjust these values to whatever you think is reasonable before giving up.